hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecord.java

```diff
@@ -107,6 +107,10 @@ public GenericRecordBuilder newBuilder() {
     @Nonnull
     @Override
     public GenericRecordBuilder newBuilderWithClone() {
+        TreeMap<String, Object> objects = new TreeMap<>();
+        for (String fieldName : getFieldNames()) {
+            objects.put(fieldName, readAny(fieldName));
+        }
         return new DeserializedGenericRecordCloner(schema, objects);
```

Comment:
```suggestion
        return new DeserializedGenericRecordCloner(schema, new TreeMap<>(objects));
```

Reply:
Then I think we should apply this to the CompactInternalGenericRecord as well. I copied this code from there.

Reply:
Sorry it does not have objects field, ignore my prev comment